plutus-core/plutus-core/src/PlutusCore/Builtin/Polymorphism.hs

```diff
@@ -173,6 +179,40 @@ data family TyAppRep (fun :: dom -> cod) (arg :: dom) :: cod
 -- | Representation of of an intrinsically-kinded universal quantifier: a bound name and a body.
 data family TyForallRep (name :: TyNameRep kind) (a :: GHC.Type) :: GHC.Type
 
+-- | For annotating an uninstantiated built-in type, so that it gets handled by the right instance
```

Comment:
Is this all here to deal with the thing that required the `newtype`, so it's kind of orthogonal to the rest of the stuff in this PR?  If I remember correctly we had a type that was actually some other type being instantiated at a phantom type and that was being processed by `DefaultUniApply`.  It might be worth mentioning that situation more explicitly here.

Reply:
> If I remember correctly we had a type that was actually some other type being instantiated at a phantom type and that was being processed by `DefaultUniApply`.

Yes, there's a [solution](https://github.com/input-output-hk/plutus/commit/8c41df16e75dfa6da5ab059244175ebfa28333b8) to that problem now with the changes this PR brings, but it's not going to be merged, because separating the universes will allow us to have a much simpler solution. And then we'll be able to remove all the `newtype` nonsense, provided it's safe to do so (no custom instances conflicting with the underlying types etc).

Reply:
> Is this all here to deal with the thing that required the `newtype`, so it's kind of orthogonal to the rest of the stuff in this PR? 

Yes, we could've preserved the hardcoded `ElaborateBuiltin` (given that the unhardcoded one does 100% the same thing for `DefaultUni` anyway), but it was an easy change, so I shoved it in too. It was nonsensical to hardcode a certain way of elaborating built-in types for all universes to begin with. But that's the general story of the builtins machinery: things are stupidly hardcoded until I get some spare time to fix that and make the machinery more flexible (and sometimes simpler at the same time).

I guess we could've just hardcoded `DefaultUni` everywhere without implementing any extensible built-in types. I do think having extensible built-in _functions_ pays off very well, but extensible built-in types seem useless until someone else decides to use Plutus on their chain or something (which is how extensible built-in types exist IIRC with the idea of running Plutus on Midnight).